### ISSUE TITLE
[3.5] Bug in book navigation

### ISSUE THREAD

**Issue by Toflar** [contao/core#8104]
The book navigation displays the links only if there's an `href`:

```html
    <?php if ($this->prevHref): ?>
    <li class="prev"><a href="<?= $this->prevHref ?>" title="<?= $this->prevPageTitle ?>">&lt; <?= $this->prevLink ?></a></li>
    <?php else: ?>
    <li class="prev empty">&nbsp;</li>
    <?php endif; ?>

    <?php if ($this->nextHref): ?>
    <li class="next"><a href="<?= $this->nextHref ?>" title="<?= $this->nextPageTitle ?>"><?= $this->nextLink ?> &gt;</a></li>
    <?php else: ?>
    <li class="next empty">&nbsp;</li>
    <?php endif; ?>
```

However, both of them can be empty if e.g. you have a `Home` site with the alias `index`. In that case `generateFrontendUrl()` will return an empty string and thus, the link is not displayed.

**Comment by leofeyer** [comment]
The correct fix would be to set the `href` to `./` then, just like [we did in the navigation menu](https://github.com/contao/core/issues/7967), wouldn't it?

**Comment by Toflar** [comment]
No, correct would be to introduce 'hasPrev' as well as 'hasNext' variables, imho. :)

**Comment by leofeyer** [comment]
That would not change anything, because you would still have to check for `$this->prevHref` to know whether to render the `<a>` tag or not.

Also, what is the point of adding an empty `<li>` tag? That seems weird to me.

**Comment by leofeyer** [comment]
Additionally, how can `$this->nextHref` ever be empty?

**Comment by Toflar** [comment]
> That would not change anything, because you would still have to check for $this->prevHref to know whether to render the <a> tag or not.

No, that's exactly the point. If `prevHref` is empty, it might be correct because empty means root :-)

> Also, what is the point of adding an empty <li> tag? That seems weird to me.

No idea.

> Additionally, how can $this->nextHref ever be empty?

Same as `prevHref`...if the Alias of the `nextHref` is empty. Obviously that's only a theoretical case for the core but e.g. I extended the core book navigation to add a rotating feature (on the last page, you can jump to the first again) and that's where `nextHref` would be `''` as well :)

**Comment by leofeyer** [comment]
Fixed in 856d7eb6fa8a468695f37e96bd86eca036ef8888.
